server/server/src/metrics.rs

```diff
@@ -0,0 +1,221 @@
+use crate::keys::public_key;
+use crate::{keys, ServerError, ServerState};
+use lazy_static::lazy_static;
+use libsecp256k1::PublicKey;
+use lockbook_models::api::FileUsage;
+use lockbook_models::file_metadata::{EncryptedFileMetadata, FileType};
+use log::error;
+use prometheus::{register_int_gauge_vec, IntGaugeVec};
+use prometheus_static_metric::make_static_metric;
+use redis::{AsyncCommands, AsyncIter};
+use redis_utils::converters::JsonGet;
+use std::collections::HashSet;
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+use uuid::Uuid;
+
+make_static_metric! {
+    pub struct MetricsStatistics: IntGauge {
+        "type" => {
+            total_users,
+            active_users,
+            total_documents,
+            total_document_bytes,
+        },
+    }
+}
+
+lazy_static! {
+    pub static ref METRICS_COUNTERS_VEC: IntGaugeVec = register_int_gauge_vec!(
+        "lockbook_metrics_counters",
+        "Lockbook's basic metrics of users and files derived from redis.",
+        &["type"]
+    )
+    .unwrap();
+    pub static ref METRICS_STATISTICS: MetricsStatistics =
+        MetricsStatistics::from(&METRICS_COUNTERS_VEC);
+}
+
+// # required metrics
+// X total users
+// active percent (just active users / total users in metrics.lockbook.com)
+// active users
+// new users
+// X total documents
+// X total bytes
+// premium users and percentage
+
+#[derive(Debug)]
+pub enum MetricsError {}
+
+// pub struct MetricsPreProcessingInfo<'a> {
+//     pub public_keys: Vec<PublicKey>,
+//     pub ids: HashMap<&'a PublicKey, Uuid>,
+//     pub metadatas: HashMap<&'a PublicKey, EncryptedFileMetadata>
+// }
+
+pub const TWO_DAYS_IN_MILLIS: u128 = 172800000;
+
+pub async fn init(server_state: &Arc<ServerState>) {
```

Comment:
+ This doesn't actually have to be `async`.

+ `init`, is a missed opportunity here imo, you should call this `start_metrics_worker`.